src/main/java/com/myhexin/cragent/coderabbittest/FofMqForSendEmailService.java

```diff
@@ -0,0 +1,104 @@
+package com.myhexin.cragent.coderabbittest;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+@Slf4j
+@Component
+public class FofMqForSendEmailService {
+
+    private TTradeTradereq dealSuper(PdfDto pdfDto, TTradeTradereq tradeReq) {
+        // 判断当前交易单号是否属于超级转换
+        SuperConvertSplitRef splitRef = superConvertSplitRefManager.querySuperConvertSplitRef(pdfDto.getTTradeTradereq().getVcAppsheetserialno());
+        if (null != splitRef) {
+            // 如果是超级转换，增加备注
+            // 如果是第一笔赎回+申购，查询申购信息
+            if (SystemConstant.ONE.equals(splitRef.getOriginalFlag())) {
+                String value = fofProductPropertyService.findValue(tradeReq.getVcCustid(), BizParamDefine.SUPER_CONVERT_DOCUMENT_MODE_PROPERTY);
+                SuperConvertDocumentModeEnum mode = SuperConvertDocumentModeEnum.of(value);
+                if (mode == SuperConvertDocumentModeEnum.MODE_136_142) {
+                    // 这种模式下，只有超转024会进这里
+                    if (!TradeTypeEnum.SHU_HUI.getCode().equals(splitRef.getBusinessCode())) {
+                        throw new IllegalStateException("#setTradeReq error. unexpected 136+142"
```

Comment:
**Issue**: The exception message "#setTradeReq error. unexpected 136+142" is hardcoded and repeated, which can lead to inconsistencies in error handling and messages. <br> **Fix**: Define all error messages as constants in a separate utility class or configuration file. This approach centralizes message management and eases future modifications or localization. <br> **Code Suggestion**: 
 ```
 - throw new IllegalStateException("#setTradeReq error. unexpected 136+142"
 + throw new IllegalStateException(ErrorMessages.UNEXPECTED_TRADE_REQ_136_142)
 ```